graylog2-server/src/test/java/org/graylog2/contentpacks/facades/ViewFacadeTest.java

```diff
@@ -0,0 +1,291 @@
+
+import static com.lordofthejars.nosqlunit.mongodb.InMemoryMongoDb.InMemoryMongoRuleBuilder.newInMemoryMongoDbRule;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ViewFacadeTest {
+
+
+    @ClassRule
+    public static final InMemoryMongoDb IN_MEMORY_MONGO_DB = newInMemoryMongoDbRule().build();
+    @Rule
+    public final MongoConnectionRule mongoRule = MongoConnectionRule.build("test");
+
+    private final ObjectMapper objectMapper = new ObjectMapperProvider().get();
+
+    public static class TestSearchDBService extends SearchDbService {
+        protected TestSearchDBService(MongoConnection mongoConnection,
+                                      MongoJackObjectMapperProvider mapper) {
+            super(mongoConnection, mapper, dto -> new SearchRequirements(Collections.emptySet(), dto));
+        }
+    }
+
+    public static class TestViewService extends ViewService {
+        protected TestViewService(MongoConnection mongoConnection,
+                                  MongoJackObjectMapperProvider mapper,
+                                  ClusterConfigService clusterConfigService) {
+            super(mongoConnection, mapper, clusterConfigService,
+                    dto -> new ViewRequirements(Collections.emptySet(), dto));
+        }
+    }
+
+    private ViewFacade facade;
+    private TestViewService viewService;
+    private TestSearchDBService searchDbService;
+    private final String viewId = "5def958063303ae5f68eccae"; /* stored in database */
+    private final String newViewId = "5def958063303ae5f68edead";
+    private final String newStreamId = "5def958063303ae5f68ebeaf";
+    private final String streamId = "5cdab2293d27467fbe9e8a72"; /* stored in database */
+
+
+    @Before
+    public void setUp() {
+        objectMapper.registerSubtypes(new NamedType(AggregationConfigDTO.class, AggregationConfigDTO.NAME));
+        objectMapper.registerSubtypes(new NamedType(MessageListConfigDTO.class, MessageListConfigDTO.NAME));
+        objectMapper.registerSubtypes(new NamedType(TimeHistogramConfigDTO.class, TimeHistogramConfigDTO.NAME));
+        objectMapper.registerSubtypes(new NamedType(OrFilter.class, OrFilter.NAME));
+        objectMapper.registerSubtypes(new NamedType(StreamFilter.class, StreamFilter.NAME));
+        objectMapper.registerSubtypes(new NamedType(QueryStringFilter.class, QueryStringFilter.NAME));
+        objectMapper.registerSubtypes(new NamedType(AutoIntervalDTO.class, AutoIntervalDTO.type));
+        searchDbService = new TestSearchDBService(mongoRule.getMongoConnection(),
+                new MongoJackObjectMapperProvider(objectMapper));
+        viewService = new TestViewService(mongoRule.getMongoConnection(),
+                new MongoJackObjectMapperProvider(objectMapper), null);
+        facade = new SearchFacade(objectMapper, searchDbService, viewService);
+    }
+
+    @Test
+    @UsingDataSet(loadStrategy = LoadStrategyEnum.CLEAN_INSERT)
+    public void exportEntity() {
```

Comment:
Instead of naming tests the same as methods I would suggest to encode the expected preconditions and outcomes. In this case: `correctlyExportsExistingView` or something. That would also make it easy to spot that there is no test for the error case that the expected view doesn't exist.